### PR TITLE
Fix SDL2: use GLM to restore functionality.

### PR DESCRIPTION
**Pull request by Deledrius**
This fixes compilation with SDL2, and uses GLM to replace deprecated functions and provide GL math features going forward.

### CODE REVIEW

**Comment by zrax** [comment]
I would much rather see glm not be bundled with imaginaryMyst...  For one, it's very large and appears to still be actively developed, and two, it appears to be something that's commonly available in common package repos (I checked Arch, Ubuntu, and MSYS2 -- all three have it in their main repos).

**Comment by Deledrius** [comment]
Removed the in-tree glm, and added FindGLM.cmake.

File: CMakeLists.txt [hunk 1]
```diff
@@ -61,7 +68,8 @@ include_directories(${STRING_THEORY_INCLUDE_DIRS})
 feature_summary(WHAT ALL FATAL_ON_MISSING_REQUIRED_PACKAGES)
 include_directories(${GLEW_INCLUDE_DIR})
 include_directories(${OPENGL_INCLUDE_DIR})
-include_directories(${SDL2_INCLUDE_DIRS})
+include_directories(${SDL2_INCLUDE_DIR})
```
Comment (zrax) [inline CMakeLists.txt]: Is this still needed when using the import targets (`target_link_libraries( ... SDL2::SDL2)`)?  Often those include setting the lib's include path...

File: cmake/FindGLM.cmake [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+	find_path(
+		GLM_INCLUDE_DIR
+		NAMES glm/glm.hpp
+		PATHS
+		/usr/include
+		/usr/local/include
+		/sw/include
+		/opt/local/include
+		${GLM_ROOT_DIR}/include
+		DOC "The directory where glm/glm.hpp resides")
```
Comment (zrax) [inline cmake/FindGLM.cmake]: Spaces instead of tabs, please

File: src/Main.cpp [hunk 3]
```diff
@@ -195,7 +204,7 @@ int main(int argc, char *argv[])
 
     SDL_GL_SwapWindow(s_display);
     SDL_Delay(3000);
-    SDL_GL_DeleteContext(glContext);
```
Comment (zrax) [inline src/Main.cpp]: I don't see any evidence that SDL_DestroyWindow also deletes the GL context...  Am I missing something?

**Review by zrax — CHANGES_REQUESTED** [review]

File: src/Main.cpp (hunk 3 above)
Comment (Deledrius) [reply src/Main.cpp]: Probably not.  I think this was an error I made conveniently masked by our currently-simplistic main function.

File: CMakeLists.txt (hunk 1 above)
Comment (Deledrius) [reply CMakeLists.txt]: Yup!  Good call.  I just had to add it individually to the projects which use it.

File: cmake/FindGLM.cmake (hunk 2 above)
Comment (Deledrius) [reply cmake/FindGLM.cmake]: 👍

**Comment by Deledrius** [comment]
Requested changes have been made. :)

**Review by zrax — APPROVED** [review]

**Pull request merged**